### PR TITLE
[Modules] Added azureMonitorProfile to allow us to enable the Prometheus collector

### PR DESCRIPTION
**Pull request by AlexanderSehr**
Based on #4329 
cc: @aadev1, @JPEasier

AVM ref: https://github.com/Azure/bicep-registry-modules/pull/665

# Description
Added azureMonitorProfile to allow us to enable the Prometheus collector and kube state metrics for prometheus addon profile for the container service cluster

Update azure/main.test.bicep and example in readme to test enabling Prometheus collector

This change will allow enabling the Prometheus collector and kube state metrics for prometheus addon profile for the container service cluster by adding the settings required to theazureMonitorProfile as per the 'Microsoft.ContainerService/managedClusters' documentation. See link below:

https://learn.microsoft.com/en-us/azure/templates/microsoft.containerservice/2023-07-02-preview/managedclusters?pivots=deployment-language-bicep#managedclusterazuremonitorprofilemetrics

| Pipeline |
| - |
| [ContainerService - ManagedClusters](https://github.com/Azure/ResourceModules/actions/workflows/ms.containerservice.managedclusters.yml) |

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

**Comment by aadev1** [comment]
@AlexanderSehr - Do you require any further changes from me or is this PR just pending review/approval?

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr - Do you require any further changes from me or is this PR just pending review/approval?

Hey @aadev1 no, thank you :) I just need one of the other maintainers to approve the PR for me @Azure/module-maintainers

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**
